src/collector.rs

```diff
@@ -0,0 +1,81 @@
+/// General-purpose garbage collector.
```

Comment:
"General-purpose" appears several times in this file. Usually I think of general-purpose GCs as those that are integrated into programming languages (e.g. JVM's GC). Perhaps this should read as "Epoch-based garbage collector"?